src/InfiniteIterableValue.php

```diff
@@ -231,12 +253,12 @@ public function slice(int $offset, int $length): InfiniteIterableValue
              * @phpstan-return iterable<TKey, TValue>
              */
             static function (iterable $iterable) use ($offset, $length): iterable {
-                foreach ($iterable as $value) {
+                foreach ($iterable as $key => $value) {
                     if ($offset-- > 0) {
                         continue;
                     }
 
-                    yield $value;
+                    yield $key => $value;
```

Comment:
Seems like it will yield 1 element when initial `$length` is 0 or less?

Reply:
I think it should return empty iterator for length = 0